Sources/Einstein/Shared/Image/ImageCache.swift

```diff
@@ -0,0 +1,61 @@
+//
+//  ImageCache.swift
+//  
+//
+//  Created by Richard Blanchard on 6/23/19.
+//
+
+#if os(iOS)
+import Combine
+import Foundation
+import UIKit
+
+public class ImageCache {
+    
+    public enum Error: Swift.Error {
+        case couldNotConvertData(URL)
+        case general(Swift.Error )
+        case unknown
+    }
+    
+    private var imageMap: [URL: UIImage] = [:]
+    private let dispatchQueue = DispatchQueue(from: StringKey(rawValue: "com.ImageCache"))
+    
+    public subscript(key: URL) -> Publishers.Future<UIImage, Error>? {
+        get {
+            let imageForKey = image(for: key)
+            _ = imageForKey
+                .sink(receiveValue: { [weak self, key] image in
+                    self?.imageMap[key] = image
```

Comment:
I'm really bad about thread issues. Does this need a DispatchQueue around it to be done on a serial queue?